src/pcre2.h.in

```diff
@@ -464,6 +464,19 @@ released, the numbers must not be changed. */
 #define PCRE2_CONFIG_COMPILED_WIDTHS        14
 #define PCRE2_CONFIG_TABLES_LENGTH          15
 
+/* Optimization directives for pcre2_set_optimize().
+For binary compatibility, only add to this list; do not renumber. */
+
+#define PCRE2_OPTIMIZATION_NONE   0
+#define PCRE2_OPTIMIZATION_FULL   1
+
+#define PCRE2_DO_AUTO_POSSESS     (1 << 8)
+#define PCRE2_DO_DOTSTAR_ANCHOR   (2 << 8)
+#define PCRE2_DO_START_OPTIMIZE   (4 << 8)
+
+#define PCRE2_AUTO_POSSESS_OFF    (1 << 8) | 1
+#define PCRE2_DOTSTAR_ANCHOR_OFF  (2 << 8) | 1
+#define PCRE2_START_OPTIMIZE_OFF  (4 << 8) | 1
```

Comment:
I like this variant much better.

However, we still exhaust the bits pretty fast. 

The naming should be consistent. Instead of PCRE2_DO_AUTO_POSSESS / PCRE2_AUTO_POSSESS_OFF 
PCRE2_DO_AUTO_POSSESS / PCRE2_DONT_AUTO_POSSESS
or
PCRE2_AUTO_POSSESS / PCRE2_AUTO_POSSESS_OFF
I have no preference, just pick one you like.

PCRE2_AUTO_POSSESS / PCRE2_NO_AUTO_POSSESS is also ok for me.

The values:
#define PCRE2_AUTO_POSSESS 64
#define PCRE2_NO_AUTO_POSSESS 65

Could be transformed to flag by:
```
if ((index & 0x1) == 0)
  flags |= 1 << ((index >> 1) - 32)
else
  flags &= ~(1 << ((index >> 1) - 32))
```

Reply:
OK, nice... so we can just set aside a range from 64-191 which will be used to switch single bits on and off. (That's in case the internal bitmap ever reaches as much as 64 bits.)

Reply:
I would say all 64+ numbers are optimization bits. A usual compiler has 5-6 generic optimization level, so I thought 64 should be more than enough for pcre2. If it seems not enough, we can simply start from a higher number, such as 256 or 1024. I would still prefer a lower number, max 128, because immediate generation on some cpus such as old arm32 systems is limited.

Reply:
Instead of having 2 `#define` it will be better to add an extra "BOOL" parameter for turning each named optimization ON/OFF. Of course using `BOOL` wouldn't be a good idea in an API so maybe since the number is no longer a bitmap either we could have a much simpler API:

```
int pcre2_set_optimize(int, int)
```